mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2848,31 +2848,75 @@ struct LLVMOpAsmDialectInterface : public OpAsmDialectInterface {
 // DialectInlinerInterface
 //===----------------------------------------------------------------------===//
 
+// Check whether the given alloca is an input to a lifetime intrinsic,
+// optionally passing through one or more casts on the way.
+static bool hasLifetimeMarkers(LLVM::AllocaOp allocaOp) {
```

Comment:
You may want to use:

`void getForwardSlice(Operation *op, SetVector<Operation *> *forwardSlice,
                     TransitiveFilter filter = nullptr /* pass-through*/);`

The filter parameter allows you to stop the computation of all transitive uses of an operation (AFAIK).